### ISSUE TITLE
TelegramClient.IsConnected is not working as expected

### ISSUE THREAD

**Issue by solarin** [sochix/TLSharp#936]
we have discussed about this, and it seemed to be fixed with the new pr #931 

But there are still some cases, when it's not working as expected. Consider this real scenario:

```
ConnectAsync
            await sender.Send(invokewithLayer, token).ConfigureAwait(false);
            await sender.Receive(invokewithLayer, token).ConfigureAwait(false);

```

Send or Receive generate an exception, which is caught at the higher level which, in turn, checks whether Client.IsConnected to eventually reconnect. 

At this point IsConnected = true, but the handhaking failed, hence it's not connected.

**Comment by VictorShap** [comment]
I have the same problem

**Comment by solarin** [comment]
> I have the same problem

yes, this should be fixed and the solution might no be easy

**Comment by VictorShap** [comment]
Okay, could you tell me, when you find the solution

**Comment by knocte** [comment]
Does #937 fix this issue?

**Comment by solarin** [comment]
yes, if you push this fix, i'll push the other fix for the choice of the data center

**Comment by knocte** [comment]
Why is #937 connected to the MediaOnly issue?

**Comment by solarin** [comment]
written in the prose

> 
> yes, filtering the MediaOnly also fixes the problem, but ReconnectToDcAsync should also be fixed because it creates a TcpTransport, then it calls ConnectAsync which basically reloads the session with the old datacenter address and recreates a new TcpTransport.
> my proposed solution is that ReconnectToDcAsync does not recreate a TcpTransport, but saves the session after updating the new datacenter. then everything stays as it is.

**Comment by solarin** [comment]
> Why is #937 connected to the MediaOnly issue?

long story short is that if i fix MediaOnly problem, it's still not working, because the new datacenter is ignored. #937 must be pushed in order for the data center selection to work properly, regardless of the MediaOnly thing

**Comment by solarin** [comment]
maybe my new pr will answer clearer the question
